src/mlpack/tests/main_tests/range_search_test.cpp

```diff
@@ -0,0 +1,292 @@
+/**
+ * @file range_search_test.cpp
+ * @author Niteya Shah
+ *
+ * Test mlpackMain() of range_search_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "RangeSearchMain";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/range_search/range_search_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct RangeSearchTestFixture
+{
+ public:
+
+  RangeSearchTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+  ~RangeSearchTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(RangeSearchMainTest, RangeSearchTestFixture);
+/*
+* Check that the correct output is returned for a small synthetic input
+* case
+*/
+BOOST_AUTO_TEST_CASE(SyntheticRangeSearch)
+{
+  cout<<"Synthetic Test 1"<<endl;
+  //Matrix Input is expected in this format
+  arma::mat x={{0, 3, 3, 4, 3, 1},{4, 4, 4, 5, 5, 2},{0, 1, 2, 2, 3, 3}};
```

Comment:
For formatting we should follow the style guide:

```
arma::mat x = {{0, 3, 3, 4, 3, 1},
               {4, 4, 4, 5, 5, 2},
               {0, 1, 2, 2, 3, 3}};
```

I did that because I think this will be over 80 columns if we put it all on one line.  Note the spaces around the `=`.  If you can take care of that all through the code it would be great.  The style guide is here: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines